ZHMModSDK/Include/IModSDK.h

```diff
@@ -68,9 +68,10 @@ class IModSDK
      * @param p_NewCode A buffer containing the new code to write to the location where the pattern was found.
      * @param p_CodeSize The size of the code buffer.
      * @param p_Offset The offset to add to the address where the pattern was found.
+	 * @param p_OldCode A (optional) buffer to store the old code.
      * @return True if the pattern was found and patched, false otherwise.
      */
-    virtual bool PatchCode(const char* p_Pattern, const char* p_Mask, void* p_NewCode, size_t p_CodeSize, ptrdiff_t p_Offset) = 0;
+    virtual bool PatchCode(const char* p_Pattern, const char* p_Mask, void* p_NewCode, size_t p_CodeSize, ptrdiff_t p_Offset, void* p_OldCode = nullptr) = 0;
```

Comment:
This is a breaking change, so I'd suggest introducing a new virtual method that does this instead (should be placed at the bottom of the list of existing virtual methods).

Reply:
Fixed, had no idea what to call the new one so `PatchCodeGetOld` is what I settled on.

Other candidates:
- `PatchCodeIWantTheOldCode`
- `GimmeOldCodeWhileChangingExistingCode`
- `OldPatchCode`
- `SenilePatchCode`